### ISSUE TITLE
Support missing deployedDomain / basePath prefix

### ISSUE THREAD

**Issue by simdevmon** [sdaschner/jaxrs-analyzer#42]
I have an application that is not bound to a single domain, but runs on multiple environments (also locally). 

SwaggerUI has the behavior that the current server address and port is used automatically, if the element `host` is missing in `swagger.json`. 

Therefore it would be useful, if the element `host` is not generated into `swagger.json`, if the `deployedDomain` element is missing, instead of using `example.com`. 

Swagger will automatically use for example `http://localhost:8080` or `http://192.168.1.2` (hostname / IP, depending on the client). 

The problem left is the application name, which is not used. The examples look like this
`http://localhost:8080/resources/pet/...` but should be
`http://localhost:8080/APPLICATION_NAME/resources/pet/...`

One possible solution would be to introduce a pastPath prefix, which is the application name. So instead of `resources` it would become `application-name/resources`. 

Edit: moved the issue to the correct project

**Comment by rmpestano** [comment]
To make it working locally I had to:

- Change swagger backend to append application root [here](https://github.com/sdaschner/jaxrs-analyzer/blob/master/src/main/java/com/sebastian_daschner/jaxrs_analyzer/backend/swagger/SwaggerBackend.java#L100)
- Put an empty string [here](https://github.com/sdaschner/jaxrs-analyzer-maven-plugin/blob/master/src/main/java/com/sebastian_daschner/jaxrs_analyzer/maven/JAXRSAnalyzerMojo.java#L198) when deployedDomain is null or empty.

I can send a PR if you agree.

**Comment by sdaschner** [comment]
You're right, due to the (latest?) Swagger spec it's allowed to leave out the `host`. I kinda breaks backwards compatibility of our tool, if somebody is relying on the host being there, but I think it's definitely the best way.

Yes of course, please PR :-)

**Comment by cristina-brinza** [comment]
Hi,

swagger UI is in my case reachable at the following endpoint (all here are examples) : localhost:12345. All the calls are against localhost:12345/myapi/myversion. However, I don't want to specify a deployedDomain, since I have the application available at different hosts (example.com ; example2.com). The plugin always appends at the moment the project name to the basePath, so swagger tries to make the curl requests agains localhost:12345/APPLICATION_NAME/myapi/myversion, which is not what I would expect. 

Could I somehow solve this issue?

Thanks! :)
